.github/workflows/build.yml

```diff
@@ -63,18 +63,26 @@ jobs:
       - uses: actions/checkout@v2
 
       - name: Use Python 3.10
-        uses: actions/setup-python@v2
+        uses: actions/setup-python@v4
         with:
           python-version: '3.10'
 
       - name: Install cibuildwheel
         run: python -m pip install cibuildwheel==2.9.0
 
       - id: set-matrix
-        run: python build-wheels-matrix.py
+        run: >
+          MATRIX=$(
+            echo $(
+              cibuildwheel --print-build-identifiers --platform linux
+                | sed 's/^.*$/{"only": "&", "os": "ubuntu-latest"}/g'
+              && cibuildwheel --print-build-identifiers --platform macOS
+                | sed 's/^.*$/{"only": "&", "os": "macos-latest" }/g'
+            ) | jq -s -c )
+          && echo ::set-output name=matrix::"$MATRIX"
```

Comment:
FYI, the environment variable or pyproject.toml variable for arch should be set at this point.